src/blog/2024/05/node-red-mind-stack-with-flowfuse.md

```diff
@@ -0,0 +1,207 @@
+
+The subsequent sections will detail system operations and describe how various components—such as temperature and humidity sensors, MQTT topics, and InfluxDB interactions—integrate within our application.
+
+### System Operation and Working Details
+
+This section outlines the operation of our system, detailing the integration of sensors, MQTT brokers, and the use of FlowFuse Dashboard 2.0 for data visualization.
+
+#### Data Collection and Publication:
+- **Sensor Integration:** A temperature and humidity sensor publishes data—temperature in degrees Celsius and relative humidity (RH) in percentage—to an MQTT broker.
+- **MQTT Subscription:** The system subscribes to topics `flowfuse/temp` and `flowfuse/RH` using the "MQTT in" node provided by FlowFuse, facilitating real-time data acquisition.
+
+#### Data Logging and Visualization:
+- **InfluxDB Logging:** A "function node" captures and logs the data into InfluxDB, ensuring that all sensor readings are stored efficiently for historical analysis.
+- **Real-time and Historical Data Visualization:** Dashboard 2.0 gadgets are employed to visualize both live and historical data. This allows for interactive engagement with the data streams.
+
+#### Interactive Data Querying:
+- **Historical Data Interface:** Users can select start and stop dates and times via a user-friendly interface. The data from this selection is processed using a function that converts it into an InfluxDB-compatible format for querying.
+- **Data Aggregation Display:** The system provides options to display maximum, minimum, and total count values for temperature and humidity. This feature enhances the user's ability to interact with the data without requiring direct queries.
+
+#### Application Structure:
+Our application architecture is designed to optimize data flow and visualization:
+- **Data Collection:** Collect data from the MQTT broker and transform it into a format suitable for logging in InfluxDB.
+- **Function Integration:** Utilize Node-RED functions to link dashboard elements with InfluxDB, facilitating dynamic data interaction.
+- **Data Visualization:** Employ Dashboard 2.0 to display data according to user-defined parameters, focusing on historical and live data obtained from sensors.
+
+#### Technical Insights:
+- **MQTT Topic Formatting:** Topics generally follow the format `topic/subtopic`. We utilize Node-RED's standard "change" node to ensure topics are reformatted into a syntax compatible with InfluxDB.
+- **Timestamps and Data Querying:** InfluxDB uses timestamps formatted per RFC3339 UTC. This standard will be crucial when querying historical data, as outlined in subsequent sections.
+
+The main objective of this project is to empower users to fully control their data with minimal prior knowledge of database systems while enjoying the enhanced UI and UX of FlowFuse Dashboard 2.0.
+
+### Development and Design Using the STORE - PROCESS - RETRIEVE - TRANSFORM Method
```

Comment:
Is this a standard? It sounds like ETL, but googling this term nothing came up. It feels like Extract Transform are the first stages though. Extract data from sensor, Transform for storage, and later Load for usage